juju/entrypoint.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+# use modules from the lib directory - works under an assumption that
+# the current directory is the top-level charm directory ($JUJU_CHARM_DIR)
+sys.path.append('lib')
+
+from juju.framework import Framework # noqa
+from charm import Charm # noqa
+
+
+# reuse the same db file as used by charm-helpers
+DB_PATH = os.path.join(
+    os.environ.get('CHARM_DIR', ''), '.unit-state.db')
+
+try:
+    from charmhelpers.core.hookenv import log
+except (ImportError, ModuleNotFoundError):
+    log = print
+
+
+def emit_charm_event(charm, event_name):
+    """Emits a charm event based on a Juju event name
+
+    :param charm: A charm instance to emit an event from
+    :type charm: py:class:`juju.charm.Charm`
+    :param str event_name: A Juju event name to emit on a charm
+    """
+    formatted_event_name = event_name.replace('-', '_')
+    try:
+        event_to_emit = getattr(charm.on, formatted_event_name)
+    except AttributeError as e:
+        msg = f"Event {formatted_event_name} not defined for {charm}"
```

Comment:
In contrast with comments, error messages should be lowercased, so that merging them together remains reasonable. You know, the kind of error that looks like this:

"Foobarer broke down: water pipes were loose: we're in trouble"

vs.

"Foobarer broke down: Water pipes were loose: We're in trouble"